### PR TITLE
Expand non-quick installation content

### PR DESCRIPTION
**Pull request by tahliar**
### PR creator: Description

~Adding a full deployment guide~ Expanding the Admin Guide installation content so that all the options are clear. This does **not** replace the Installation and Setup Quick Start, which has a different use case. 

This is an expansion of the installation content for **15 SP7**. It can be fine-tuned even further (and crossover with SAP more) in SLE 16, which has a whole new structure that makes sharing content easier. 

Here's a PDF of the Admin Guide as a whole: 
[book-administration_en.pdf](https://github.com/user-attachments/files/20007189/book-administration_en.pdf)
For the most part I made changes to Part I: Installation and setup, but I did make a few changes in other parts too. I'll add a section below that goes into greater detail about the changes, since the diff is too big to be coherent. 
See **Summary of changes** below.

This was a long-term project and therefore needed to be rebased regularly, so the commit dates in this PR do not reflect reality.

I originally kept track of the relevant product changes for each SP, but I've decided that it doesn't make sense to backport this, both because of complexity and to avoid issues with the localised content in earlier versions.

### PR creator: Are there any relevant issues/feature requests?

* jsc#PED-2842

### PR creator: Which product versions do the changes apply to?

When opening a PR, check all versions of the documentation that your PR applies to.

- SLE-HA 15
  - [x] 15 next *(current `main`, no backport necessary)*
  - [ ] 15 SP6
  - [ ] 15 SP5
  - [ ] 15 SP4
  - [ ] 15 SP3
  - [ ] 15 SP2
- SLE-HA 12
  - [ ] 12 SP5

### PR reviewer only: Have all backports been applied?

The doc team member merging your PR will take care of backporting to older documents.
When opening a PR, do *not* set the following check box.

- [ ] all necessary backports are done

# Summary of changes

### Main change: major expansion of the content in "Part I: Installation and setup"

For this I'll attach a PDF, as it's a whole Part. This PDF contains _only_ Part I: 
[part-install_en.pdf](https://github.com/user-attachments/files/20007192/part-install_en.pdf)

**New content**

- Added a new chapter "2 Installation overview". This gives an introduction to the rest of the chapters in this part, i.e. the high-level procedure for setting up a cluster.
- Added a new chapter "4 Logging in to the cluster nodes". Before installing anything, this explains the root vs sudo user options, and SSH agent forwarding.
- Added a new chapter "6 Using the bootstrap scripts". This goes into detail about `crm cluster init` and `crm cluster join`, with options beyond just the quickstart defaults. It also includes a quick section on `crm cluster remove`.
- Added a new section "7.4 Configuring an arbitrator for quorum decisions" because the QDevice window in YaST wasn't documented before.

**Edited content**

- Slightly restructured chapter "1 Product overview", including making STONITH stand out more in section "1.4 Architecture".
- Slightly restructured chapter "7 Using the YaST cluster module" to follow the initial wizard better, but moved the non-YaST parts to the end (i.e. transferring the config to all nodes and starting the cluster).
- Tested and updated all sections in chapter "7 Using the YaST cluster module".
- Moved AutoYaST to a separate chapter "8 Deploying nodes with AutoYaST". 

### Minor change: Updated/edited sections of "Part II: Configuration and administration"

This is new content to explain more of the post-installation functions of `crm cluster init` that weren't already documented. 

Note that the syntax highlighting in the HTML render is a bit odd right now, but that's a style sheet issue. 

- In section "9.5.1 User privileges and authentication":

image

- In section "9.5.11 Managing Corosync configuration":

image

### CODE REVIEW

**Comment by tahliar** [comment]
**Main change: major expansion of the content in "Part I: Installation and setup"**

For this I'll attach a PDF, as it's a whole Part. This PDF contains _only_ Part I: 
[part-install_en.pdf](https://github.com/user-attachments/files/19460744/part-install_en.pdf)

List of changes:

- Added a new chapter "2 Installation overview". This gives an introduction to the rest of the chapters in this part, i.e. the high-level procedure for setting up a cluster.
- Added a new chapter "4 Logging in to the cluster nodes". Before installing anything, this explains the root vs sudo user options, and SSH agent forwarding.
- Added a new chapter "6 Using the bootstrap scripts". This goes into detail about `crm cluster init` and `crm cluster join`, with options beyond just the quickstart defaults. It also includes a quick section on `crm cluster remove`.
- Slightly restructured chapter "7 Using the YaST cluster module" to follow the initial wizard better, but moved the non-YaST parts to the end (i.e. transferring the config to all nodes and starting the cluster).
- Added a new section "7.4 Configuring an arbitrator for quorum decisions" because the QDevice window wasn't documented before.

**Review by gao-yan — COMMENTED** [review]
Thanks for the nice work, @tahliar !

Some comments in below.

@zzhou1 @liangxin1300 , you definitely want go through this.

File: xml/ha_installation_overview.xml [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+            A physical device (a power switch). Each node must have at least one device.
+            We recommend multiple &stonith; devices per node. To configure the cluster to
+            use physical &stonith; devices, see <xref linkend="cha-ha-fencing"/>.
+          </para>
+        </listitem>
+        <listitem>
+          <para>
+            SBD (&stonith; Block Device) in combination with a watchdog. SBD can be used with
+            shared storage or in diskless mode. Diskless SBD should only be used in clusters
+            with more than two nodes. For disk-based SBD, we recommend multiple SBD devices.
```
[the remark this answers is not in the document]
Comment (tahliar) [reply xml/ha_installation_overview.xml:120]: That's definitely a good point, but might get a bit wordy for this overview. I'll mention it more prominently in the SBD chapter.

File: xml/ha_config_cli.xml [hunk 2]
```diff
@@ -954,6 +968,88 @@ Membership information
...
+<screen>&prompt.root;<command>crm cluster join -c <replaceable>NODE1</replaceable></command></screen>
+          <para>
+            Accept the proposed network address.
+          </para>
+        </listitem>
+        <listitem>
+          <para>
+            For two rings, specify the same network interfaces you used for the first node:
+          </para>
+<screen>&prompt.root;<command>crm cluster join -c <replaceable>NODE1</replaceable> -i eth0 -i eth1</command></screen>
```
[the remark this answers is not in the document]
Comment (liangxin1300) [reply xml/ha_config_cli.xml:1042]: Yes, when init node using two rings, the join node must specify two interfaces

File: xml/ha_installation_overview.xml [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+        <listitem>
+          <para>
+            <xref linkend="cha-ha-ycluster"/>
+          </para>
+        </listitem>
+      </itemizedlist>
+      <para>
+        <xref linkend="tab-ha-installation-overview"/> shows the components that can be configured
+        by each method. You can also use a combination of methods. For example, you could set up
+        one node with &yast;, then use the bootstrap scripts to add more nodes. You can also clone
```
[the remark this answers is not in the document]
Comment (liangxin1300) [reply xml/ha_installation_overview.xml:96]: This part of doc means, the first node using yast, then other nodes using `crm cluster join`? 
Please don't do that!
We should always recommend user to either use yast or use crmsh bootstrap on all nodes, don't mix them!

File: xml/ha_bootstrap_install.xml [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
+          </para>
+        </listitem>
+        <listitem>
+          <para>
+            To use disk-based SBD, enter a persistent path to the partition of the block device you
+            want to use. The path must be consistent across all nodes in the cluster, for example,
+            <filename>/dev/disk/by-id/<replaceable>ID</replaceable></filename>.
+          </para>
+          <para>
+            The script creates a small partition on the device to be used for SBD.
```
[the remark this answers is not in the document]
Comment (liangxin1300) [reply xml/ha_bootstrap_install.xml:319]: If the device is already initialized by `sbd create`:
In interactive mode, crmsh will ask:
```
Do you wish to use SBD (y/n)? y
Path to storage device (e.g. /dev/disk/by-id/...), or "none" for diskless sbd, use ";" as separator for multi path []/dev/sda5
SBD is already configured to use /dev/sda5 - overwrite (y/n)?
```
If choose `n`, the metadata in sbd device wouldn't be overwritten;
If choose `y`, the metadata in sbd device will be overwritten by the defined parameter in /etc/crm/profiles.yml

In non-interactive mode(crm cluster init -s <device> -y), the metadata in sbd device will also be overwritten

File: xml/ha_installation_overview.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (tahliar) [reply xml/ha_installation_overview.xml:96]: Oh, thanks for this! The suggestion that you can use a mixed approach was already in the docs all the way back to 12 SP5 and probably earlier. I'll remove it everywhere.

File: xml/ha_config_cli.xml [hunk 5]
```diff
@@ -954,6 +968,88 @@ Membership information
...
+        On the first node, change &corosync;'s ring configuration with one of the
+        following commands:
+      </para>
+      <itemizedlist>
+        <listitem>
+          <para>
+            To configure one ring, run the script's <option>corosync</option> stage
+            with no additional parameters:
+          </para>
+<screen>&prompt.root;<command>crm cluster init corosync</command></screen>
```
[the remark this answers is not in the document]
Comment (tahliar) [reply xml/ha_config_cli.xml:997]: Yep, when I set up a bond and run `crm cluster init corosync`, it mentions bond0:

image

But when I don't have a bond configured, it mentions eth0:

image

File: xml/ha_config_cli.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tahliar) [reply xml/ha_config_cli.xml:1042]: Thanks!

File: xml/ha_bootstrap_install.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (liangxin1300) [reply xml/ha_bootstrap_install.xml:319]: No, crmsh don't create a disk partition

[the remark this answers is not in the document]
Comment (tahliar) [reply xml/ha_bootstrap_install.xml:319]: Looks like this statement also goes back to SLE 12: https://documentation.suse.com/sle-ha/12-SP5/html/SLE-HA-all/art-ha-install-quick.html#sec-ha-inst-quick-sbd-req

I'll backport this change too, then. Thanks!

File: xml/ha_yast_cluster.xml [hunk 6]
```diff
@@ -822,186 +1165,60 @@
...
-   </para>
-   <para>
-    This requires the following basic steps:
+    After the cluster configuration with &yast; is complete, use <command>csync2</command>
+    to copy the configuration files to the rest of the cluster nodes. To receive the files,
+    nodes must be included in the <guimenu>Sync Host</guimenu> group you configured in
+    <xref linkend="pro-ha-installation-setup-csync2-yast"/>.
    </para>
+   <para> Before running &csync; for the first time, you need to make the
+    following preparations: </para>
```
Comment (zzhou1) [inline xml/ha_yast_cluster.xml:1187]: Worth to share another simplest one is `crm cluster init -y csync2` which basically implement "PROCEDURE 7.10"

**Review by zzhou1 — CHANGES_REQUESTED** [review]

File: xml/ha_yast_cluster.xml [hunk 7]
```diff
@@ -667,12 +866,136 @@
...
+      it must be copied manually to all members of the cluster. </para>
+    </step>
+    <step>
+     <para> To populate the <guimenu>Sync File</guimenu> list with the files
+      that usually need to be synchronized among all nodes, select <guimenu>Add
+       Suggested Files</guimenu>. </para>
+    </step>
+    <step>
+     <para> To <guimenu>Edit</guimenu>, <guimenu>Add</guimenu> or
+       <guimenu>Remove</guimenu> files from the list of files to be synchronized
```
Comment (dariavladykina) [inline xml/ha_yast_cluster.xml:966]:
```suggestion
       <guimenu>Remove</guimenu> files from the list of files to be synchronized,
```

**Review by dariavladykina — COMMENTED** [review]
LGTM, only minor suggestions for the entire thing. Thanks!

File: xml/ha_yast_cluster.xml (hunk 6 above)
Comment (tahliar) [reply xml/ha_yast_cluster.xml:1187]: Ohh, thanks, that's much easier!

File: xml/ha_bootstrap_install.xml [hunk 8]
```diff
@@ -0,0 +1,517 @@
...
+  <!-- Initially duplicated from xml/article_installation.xml, heavily expanded -->
+    <title>Setting up the first node with <command>crm cluster init</command></title>
+    <para>
+      Setting up the first node with the <command>crm cluster init</command> script
+      requires only a minimum of time and manual intervention.
+    </para>
+    <para>
+      This procedure describes multiple configuration options. We recommend reviewing the
+      whole procedure before starting the script. Alternatively, for a minimal setup with
+      only the default options, see <xref linkend="article-installation"/>.
```
[the remark this answers is not in the document]
Comment (tahliar) [reply xml/ha_bootstrap_install.xml:145]: AFAIK doing an xref like this to other HA docs only creates a clickable link in the chunked HTML version, which uses the DC file SLE-HA-all. Otherwise it only resolves as a title in HTML-single and PDF (e.g. in the published version, see https://documentation.suse.com/sle-ha/15-SP6/single-html/SLE-HA-administration/#cha-ha-ycluster vs https://documentation.suse.com/sle-ha/15-SP6/html/SLE-HA-all/cha-ha-ycluster.html). 

I sometimes get around this by using a hard link instead.

File: xml/ha_bootstrap_install.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (tahliar) [reply xml/ha_bootstrap_install.xml:319]: @gao-yan @liangxin1300 

Hmm, this is also mentioned in the SBD chapter: https://documentation.suse.com/sle-ha/15-SP6/html/SLE-HA-all/cha-ha-storage-protect.html#id-1.4.4.11.3.6

I found this commit from 10 years ago: https://github.com/SUSE/doc-sleha/commit/ad62f1319dd0d668f6f0265853e1476d4f48daee
But even then the partition is already mentioned, so it's very old indeed. 

Is this something that used to be true but isn't anymore? Or am I confusing two different things?

[the remark this answers is not in the document]
Comment (tahliar) [reply xml/ha_bootstrap_install.xml:320]: I think the doc might be mixing up disk partitions and node slots? Whatever the case, I'll take the partition line out of the `crm cluster init` sections for all versions, since it doesn't really need to be there.

[the remark this answers is not in the document]
Comment (gao-yan) [reply xml/ha_bootstrap_install.xml]: > @gao-yan @liangxin1300
> 
> Hmm, this is also mentioned in the SBD chapter: https://documentation.suse.com/sle-ha/15-SP6/html/SLE-HA-all/cha-ha-storage-protect.html#id-1.4.4.11.3.6
> 
> I found this commit from 10 years ago: [ad62f13](https://github.com/SUSE/doc-sleha/commit/ad62f1319dd0d668f6f0265853e1476d4f48daee) But even then the partition is already mentioned, so it's very old indeed.
> 
> Is this something that used to be true but isn't anymore? Or am I confusing two different things?

Indeed I can tell the description there has been misleading.

We'd better change "SBD partition" to "SBD device" meanwhile probably telling something like this instead:

 "In an environment where all nodes have access to shared storage, a small logic unit or a small partition on a logic unit is formatted for use with SBD. The size of the device ..."

?

File: xml/ha_config_cli.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (gao-yan) [reply xml/ha_config_cli.xml:1042]: Since "redundant communication channel" is mentioned in the section "Setting up the first node with crm cluster init", should this be mentioned in the section "Adding nodes with crm cluster join" too, given that specifications are necessary for the case?

File: xml/ha_bootstrap_install.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (tahliar) [reply xml/ha_bootstrap_install.xml]: Thanks @gao-yan, that sounds better. I'll make that change for all versions.

File: xml/ha_config_cli.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tahliar) [reply xml/ha_config_cli.xml:1042]: Definitely! I should have thought of that, thanks for noticing!

File: xml/ha_bootstrap_install.xml [hunk 9]
```diff
@@ -0,0 +1,517 @@
...
+        </para>
+      </listitem>
+    </varlistentry>
+    <varlistentry>
+      <term>Cluster name</term>
+      <listitem>
+        <para>
+          Defines a name for the cluster, by default <systemitem>hacluster</systemitem>. This is
+          optional and mostly useful for &geo; clusters. Usually, the cluster name reflects the
+          geographical location and makes it easier to distinguish a site inside a &geo; cluster.
```
[the remark this answers is not in the document]
Comment (zzhou1) [reply xml/ha_bootstrap_install.xml:102]: Ah, thank Yan to bring up this topic. I don't have the chance to dive into this topic much. Apparently, yes, cluster_name is required for DLM. 

For corosync alone, cluster_name gives the convenience to generate multicast address for corosync.

File: xml/ha_config_cli.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tahliar) [reply xml/ha_config_cli.xml:1042]: I've added more options for the `join` script: 

image

**Review by gao-yan — APPROVED** [review]
LGTM. Awesome work, @tahliar !

**Review by dariavladykina — APPROVED** [review]

**Review by dariavladykina — APPROVED** [review]

**Review by dariavladykina — APPROVED** [review]
